stdlib/universe/fill.go

```diff
@@ -255,14 +261,32 @@ func (t *fillTransformation) fillTable(w *table.StreamWriter, cr flux.ColReader,
 		return nil
 	}
 	vs := make([]array.Interface, len(w.Cols()))
+	var colArr array.Interface
 	for i, col := range w.Cols() {
-		arr := table.Values(cr, i)
-		if i != colIdx {
-			vs[i] = arr
+
+		isMissingCol := colIdx < 0 && col.Label == t.spec.Column
+
+		if !isMissingCol {
+			colArr = table.Values(cr, i)
+		} else {
+			colArr = t.addNullColumn(flux.ColumnType(t.spec.Value.Type()), cr.Len())
+		}
+
+		if i == colIdx || isMissingCol {
+			vs[i] = t.fillColumn(col.Type, colArr, fillValue)
+		} else {
+			vs[i] = colArr
 			vs[i].Retain()
-			continue
 		}
-		vs[i] = t.fillColumn(col.Type, arr, fillValue)
 	}
 	return w.Write(vs)
 }
+
+func (t *fillTransformation) addNullColumn(typ flux.ColType, l int) array.Interface {
+	builder := arrow.NewBuilder(typ, memory.DefaultAllocator)
```

Comment:
Can you use the allocator associated with the fill transformation? It's probably `t.alloc`.